taichi/ir/frontend_ir.cpp

```diff
@@ -173,6 +173,29 @@ void UnaryOpExpression::flatten(FlattenContext *ctx) {
   ctx->push_back(std::move(unary));
 }
 
+Expr to_broadcast_tensor(const Expr &elt, const DataType &dt) {
+  TI_ASSERT(dt->is<TensorType>());
+  auto tensor_type = dt->as<TensorType>();
+  auto elt_type = tensor_type->get_element_type();
```

Comment:
Check if `elt` has the same TensorType as `dt`, then we should directly return `elt` here. Otherwise an error should be thrown.